crates/uv/src/commands/project/init.rs

```diff
@@ -77,6 +75,94 @@ pub(crate) async fn init(
         }
     };
 
+    if r#virtual {
+        init_virtual_workspace(&path, isolated)?;
+    } else {
+        init_project(
+            &path,
+            &name,
+            no_readme,
+            python,
+            isolated,
+            python_preference,
+            python_fetch,
+            connectivity,
+            native_tls,
+            cache,
+            printer,
+        )
+        .await?;
+    }
+
+    // Create the `README.md` if it does not already exist.
+    if !no_readme {
+        let readme = path.join("README.md");
+        if !readme.exists() {
+            fs_err::write(readme, String::new())?;
+        }
+    }
+
+    let project = if r#virtual { "workspace" } else { "project" };
+    match explicit_path {
+        // Initialized a project in the current directory.
+        None => {
+            writeln!(
+                printer.stderr(),
+                "Initialized {} `{}`",
+                project,
+                name.cyan()
+            )?;
+        }
+        // Initialized a project in the given directory.
+        Some(path) => {
+            let path = path
+                .simple_canonicalize()
+                .unwrap_or_else(|_| path.simplified().to_path_buf());
+
+            writeln!(
+                printer.stderr(),
+                "Initialized {} `{}` at `{}`",
+                project,
+                name.cyan(),
+                path.display().cyan()
+            )?;
+        }
+    }
+
+    Ok(ExitStatus::Success)
+}
+
+fn init_virtual_workspace(path: &Path, isolated: bool) -> Result<()> {
+    // Check nested workspaces.
+    if !isolated {
+        check_nested_workspaces(path, &DiscoveryOptions::default());
```

Comment:
Should we warn or abort if nested workspace detected?

Reply:
I think warn is ok.